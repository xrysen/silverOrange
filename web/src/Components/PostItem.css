.post-title:hover {
  cursor: pointer;
  color: white;
  background-color: #634B66;
  transition: background-color, 0.5s;
}
