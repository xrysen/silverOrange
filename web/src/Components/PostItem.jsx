export default function PostItem(props) {
  return (
    <tr>
      <td>{props.title}</td>
      <td>{props.author}</td>
      <td>{props.summary.substring(0,20) + "..."}</td>
      <td>{new Date(props.date).toISOString().substring(0, 10)}</td>
    </tr>
  );
}
