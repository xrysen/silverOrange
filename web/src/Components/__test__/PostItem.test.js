import React from 'react';
import PostItem from '../PostItem';
import renderer from 'react-test-renderer';

const post = {
  title: 'My Post',
  author: 'Alice Bob',
  summary:
    'This a test summary that is more than 60 characters long by a few characters',
  date: '2021-07-16',
};

test('Renders a table for posts and authors', () => {
  const tree = renderer.create(
    <PostItem
      title={post.title}
      author={post.author}
      summary={post.summary}
      date={post.date}
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
