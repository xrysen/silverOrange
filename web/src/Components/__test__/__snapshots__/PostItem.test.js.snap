// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a table for posts and authors 1`] = `
<tr>
  <td
    className="post-title"
  >
    My Post
  </td>
  <td>
    Alice Bob
  </td>
  <td>
    This a test summary that is more than 60 characters long by ...
  </td>
  <td>
    2021-07-16
  </td>
</tr>
`;
